src/components/VerseCheckWrapper.js

```diff
@@ -64,9 +63,16 @@ function VerseCheckWrapper({
   });
 
   useEffect(() => {
-    setLocalState({ selections });
+    // TRICKY: for async fs loads, need to update mode and selection state when new selection loads
+    setLocalState(
+      {
+        mode: getInitialMode(),
+        newSelections: selections,
+        selections,
```

Comment:
We don't need to set `selections` in the state because `newSelections` is the variable being used not selections. It looks like this used to be incorrect even before your changes.